### PR TITLE
Feat/648 use default props for angular

### LINKED ISSUE #648: useDefaultProps for Angular
we need to support defaultProps for Angular
https://github.com/BuilderIO/mitosis/pull/646

this can be as simple as
`mitosis`
```typescript
export interface Props {
  myProp: string;
}

useDefaultProps<Props>({
  myProp: 'hi'
});
default export function MyComponent(props: Props) {
}

```

`angular`
```typescript
export interface Props {
  myProp: string;
}

const defaultProps = {
  myProp: 'hi'
};

@Component({
  selector: 'my-component'
})
export default class MyComponent {
  @Input() myProp: Props["myProp"] = defaultProps["myProp"];
}
```

### PR DESCRIPTION
**Pull request by raymondmuller**
## Description

Adds default props to the angular generator.
Closes #648

### CODE REVIEW

File: packages/core/src/__tests__/__snapshots__/lit.test.ts.snap [hunk 1]
```diff
@@ -2440,13 +2440,19 @@ const spread = (properties) =>
 
 @customElement(\\"my-button\\")
 export default class Button extends LitElement {
+  createRenderRoot() {
+    return this;
+  }
```
Comment (samijaber) [inline packages/core/src/__tests__/__snapshots__/lit.test.ts.snap:2445]: Weird how this showed up in this PR's diff... 🤔 are you on the latest `main`?

Comment (raymondmuller) [reply packages/core/src/__tests__/__snapshots__/lit.test.ts.snap:2445]: Yes, but the snapshots are not updated on the main branch :)

Comment (steve8708) [reply packages/core/src/__tests__/__snapshots__/lit.test.ts.snap:2445]: ya current main is actually not passing build, prob bc two separate PRs that were each valid on their own got merged, so merging this one will fix it https://github.com/BuilderIO/mitosis/commit/48371deef473537fbf13e830d39840911f2c1956

That is the state we want the snapshots in (with the `createRenderRoot`) ^

**Review by samijaber — APPROVED** [review]
LGTM 👍🏽 

Tagging @PatrickJS in case he sees something

**Review by PatrickJS — APPROVED** [review]

**Pull request merged**